### ISSUE TITLE
[Bug report] Creating SpectrumData objects from file does not include data added with add_data() function

### ISSUE THREAD

**Issue by dkweiss31** [scqubits/scqubits#77]
I would like to add data to my `SpectrumData` object before writing it to a file, and then be able to access that data later when reading the file back to a `SpectrumData` object. Unfortunately the `SpectrumData.create_from_file()` function appears to ignore the presence of this additional data. Here is a minimal working example:

```
import numpy as np
import scqubits as qubit
from scqubits.core.storage import SpectrumData

transmon = qubit.Transmon(EJ=10., EC=1., ng=0.0, ncut=10, truncated_dim=6)

ng_list = np.linspace(-2, 2, 220)
specdata = transmon.get_spectrum_vs_paramvals('ng', ng_list, get_eigenstates=True)
specdata.add_data(test_array=np.array([1, 2, 3]))
specdata.filewrite('transmon_with_test_array.h5')

specdata_from_file = SpectrumData.create_from_file('transmon_with_test_array.h5')

specdata_from_file.test_array #throws an error
```

`specdata` has the attribute `test_array` as expected, however `specdata_from_file` does not.

OS: MacOS
scqubits: version 1.2.1
Python: 3.7

**Comment by jkochNU** [comment]
Manually added data was not properly recorded for file IO. This should be fixed with 3576e161

**Comment by dkweiss31** [comment]
Indeed it is fixed by 3576e1612703b284fd2f1ebbe84d8a13898a312d
